requirements/README.md

```diff
@@ -0,0 +1,53 @@
+# Python library requirements
+
+The files in this directory are used to create `requirements.txt` files for
+installing Python libraries via `pip`.  There are two types of files here:
+* `.in` files are unpinned requirements with very loose or unspecified version
+  numbers for the libraries.  These files include only the top-level required
+  libraries, and not their implied dependencies.
+* `.txt` files are generated from the `.in` files using `pip-compile` from
+[pip-tools](https://github.com/jazzband/pip-tools).  These are fully-resolved
+requirements files that specify an exact version for each library and all of
+their child dependencies.  It also includes a hash of the library's archive
+file, which will cause `pip` to raise a security error if the library changes
+unexpectedly.
+
+## Specifying dependencies for GitHub actions
+GitHub actions use `requirements.txt` from the root of the repository.  That
+`requirements.txt` file points to the `.in` files so it gets unpinned
+dependencies.  This causes us to always build with the latest dependencies
+available on PyPI. (DI: I personally feel it is infuriating to randomly get
```

Comment:
I'm assuming this aside is going to be removed? If there is breakage that is too bad to fix quickly (like pydantic) it's always okay to pin and create a ticket to fix the problem later.

Reply:
Well I wasn't planning to... it's there as an explanation of "why is this configured in a somewhat unusual manner".  I'll rephrase it if you find it objectionable?

Reply:
How about you do it as a footnote instead so it doesn't interrupt the flow with the complaint? For all I know github will handle `[^1]` footnote markup.